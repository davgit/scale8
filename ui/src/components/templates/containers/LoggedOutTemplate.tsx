import { FC } from 'react';
import makeStyles from '@mui/styles/makeStyles';
import LoggedOutNavigationContainer from '../../atoms/LoggedOutNavigationContainer';
import { ChildrenOnlyProps } from '../../../types/props/ChildrenOnlyProps';

const useStyles = makeStyles(() => ({
    root: {
        display: 'flex',
        flexDirection: 'column',
        minHeight: '100vh',
        minWidth: '320px',
        background: '#f9f9f9',
        position: 'relative',
    },
    section: {
        flexShrink: 0,
        width: '100%',
    },
}));

const LoggedOutTemplate: FC<ChildrenOnlyProps> = (props: ChildrenOnlyProps) => {
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <div className={classes.section}>
                <LoggedOutNavigationContainer />
            </div>
            <div className={classes.section}>{props.children}</div>
        </div>
    );
};

export default LoggedOutTemplate;
