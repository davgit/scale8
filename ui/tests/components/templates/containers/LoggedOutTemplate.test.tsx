import LoggedOutTemplate from '../../../../src/components/templates/containers/LoggedOutTemplate';
import { render } from '@testing-library/react';

jest.mock('next/router', () => ({
    useRouter() {
        return {
            route: '/',
            pathname: '',
            query: '',
            asPath: '',
            push: jest.fn(),
            events: {
                on: jest.fn(),
                off: jest.fn(),
            },
            beforePopState: jest.fn(() => null),
            prefetch: jest.fn(() => null),
        };
    },
}));

describe('LoggedOutTemplate', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('renders LoggedOutTemplate unchanged', () => {
        const { asFragment } = render(
            <LoggedOutTemplate>
                <div />
            </LoggedOutTemplate>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
