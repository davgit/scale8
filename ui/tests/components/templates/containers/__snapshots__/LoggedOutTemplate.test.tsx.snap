// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedOutTemplate renders LoggedOutTemplate unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-d97baw"
  >
    <div
      class="MuiBox-root css-sm2l8q"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-1p4tas-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1wxq31f-MuiToolbar-root"
        >
          <div
            class="MuiBox-root css-1mbqzx0"
          >
            <svg
              viewBox="0 0 186 339"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 55.03,16.48 C 76.14,-4.63 110.37,-4.63 131.48,16.48 131.48,16.48 93.25,54.69 93.25,54.69 93.25,54.69 55.03,16.48 55.03,16.48 Z"
                id="top-1"
                style="fill: #c6e54b; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
              />
              <path
                d="M 93.25,54.69 C 93.25,54.69 131.66,16.29 131.66,16.29 131.66,16.29 169.93,54.55 169.93,54.55 169.93,54.55 131.52,92.95 131.52,92.95 131.52,92.95 93.25,54.69 93.25,54.69 Z"
                id="top-2"
                style="fill: #fab300; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
              />
              <path
                d="M 169.75,54.74 C 190.86,75.84 190.86,110.06 169.75,131.16 169.75,131.16 131.52,92.95 131.52,92.95 131.52,92.95 169.75,54.74 169.75,54.74 Z"
                id="top-3"
                style="fill: #ffdc51; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
              />
              <path
                d="M 16.25,131.77 C -4.86,110.67 -4.86,76.45 16.25,55.34 16.25,55.34 54.47,93.56 54.47,93.56 54.47,93.56 16.25,131.77 16.25,131.77 Z"
                id="mid-1"
                style="fill: #01c8a1; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
              />
              <path
                d="M 16.25,131.77 C 16.25,131.77 54.65,93.38 54.65,93.38 54.65,93.38 92.92,131.64 92.92,131.64 92.92,131.64 54.52,170.03 54.52,170.03 54.52,170.03 16.25,131.77 16.25,131.77 Z"
                id="mid-2"
                style="fill: #0096a6; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
              />
              <path
                d="M 54.52,170.03 C 54.52,170.03 92.92,131.64 92.92,131.64 92.92,131.64 131.19,169.90 131.19,169.90 131.19,169.90 92.79,208.29 92.79,208.29 92.79,208.29 54.52,170.03 54.52,170.03 Z"
                id="mid-3"
                style="fill: #39cce0; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
              />
              <path
                d="M 169.46,208.16 C 169.46,208.16 131.05,246.55 131.05,246.55 131.05,246.55 92.78,208.29 92.78,208.29 92.78,208.29 131.19,169.90 131.19,169.90 131.19,169.90 169.46,208.16 169.46,208.16 Z"
                id="mid-4"
                style="fill: #aa74e6; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
              />
              <path
                d="M 169.46,208.16 C 190.57,229.26 190.57,263.48 169.46,284.59 169.46,284.59 131.24,246.37 131.24,246.37 131.24,246.37 169.46,208.16 169.46,208.16 Z"
                id="mid-5"
                style="fill: #9042e7; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
              />
              <path
                d="M 16.16,284.90 C -4.95,263.80 -4.95,229.58 16.16,208.47 16.16,208.47 54.38,246.69 54.38,246.69 54.38,246.69 16.16,284.90 16.16,284.90 Z"
                id="bottom-1"
                style="fill: #ff0084; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
              />
              <path
                d="M 92.65,284.95 C 92.65,284.95 54.25,323.35 54.25,323.35 54.25,323.35 15.98,285.08 15.98,285.08 15.98,285.08 54.38,246.69 54.38,246.69 54.38,246.69 92.65,284.95 92.65,284.95 Z"
                id="bottom-2"
                style="fill: #fe657a; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
              />
              <path
                d="M 130.87,323.16 C 109.76,344.27 75.54,344.27 54.43,323.16 54.43,323.16 92.65,284.95 92.65,284.95 92.65,284.95 130.87,323.16 130.87,323.16 Z"
                id="bottom-3"
                style="fill: #c63d51; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
              />
            </svg>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xwna19-MuiTypography-root-MuiLink-root"
              href="https://scale8.com/"
            >
              Scale8
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-sm2l8q"
    >
      <div />
    </div>
  </div>
</DocumentFragment>
`;
